### ISSUE TITLE
blank boxes: urgent

### ISSUE THREAD

**Issue by JSBLGPN** [eXistSolutions/LGPN#190]
When we have a individual with three primary sources or more, after the third one the boxes are blank! So we lose time, since we have to save, reopen and complete entries

Old problem but it needs to be fixed asap, please. 

Thanks

**Comment by tuurma** [comment]
Do I understand right, that adding a new entry, when using the + button next to Primary Sources section to add more sources, the added sections are not styled properly? 

I've just tried to add 4 primary sources and the 4th one is displayed like this (same as all previous ones)

![image](https://user-images.githubusercontent.com/449468/44091242-14c8e4fa-9fcd-11e8-8f20-6a896495e5d9.png)

It makes me wonder if the issue is not your browser. I am using Chrome browser. Could you check your browser name and version and also test if the problem occurs for you if you use recent version of Chrome?

**Comment by JSBLGPN** [comment]
I use firefox and I have just tried with Chrome Version 68.0.3440.106 (Official Build) (64-bit) and the problem is till there. 

Precision: the problem appears in boxes like source, linking and then also with place

thanks

**Comment by JSBLGPN** [comment]
Yes the problem persists. I think to see the problem you might need to complete the entire entry with genre, volume, name, date etc.

**Comment by tuurma** [comment]
Thanks to @michaelzellmann's description in #131 I was able to reproduce the issue:

re-editing existing entry, after adding new primary source field dropdown lists do not show labels or values for the options

(detailed instructions how to reproduce in https://github.com/eXistSolutions/LGPN/issues/166#issuecomment-414231739)

<img width="1032" alt="image" src="https://user-images.githubusercontent.com/449468/44311016-53478d00-a3e0-11e8-8524-342285d8864d.png">

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/449468/44311046-a6b9db00-a3e0-11e8-9dfa-b1ccf234a477.png">

**Comment by tuurma** [comment]
I have checked that records are correctly structured, also sections dynamically added into the model are correct. Interesting fact is the problem manifests itself only when specifying certain fields in added primary source section - if the ref. details box is left empty, added places will behave perfectly fine. Dropdown lists as can be seen in the screenshot above display list of entries which corresponds to the expected options in size (cf dropdown for places shows 4 entries and there are indeed supposed to be 4 options there) but labels and @value attributes are not there. Which makes me think it might unfortunately be a bug in betterforms. @JoernT would you have an advice how to proceed here?

**Comment by tuurma** [comment]
Discussing with my colleagues, we may need to restructure the forms in the future to make them easier to maintain long term.
